btcutil/psbt/utils.go

```diff
@@ -398,6 +398,27 @@ func VerifyInputOutputLen(packet *Packet, needInputs, needOutputs bool) error {
 	return nil
 }
 
+// InputsReadyToSign makes sure that all input data have the previous output
+// specified meaning that either nonwitness UTXO or the witness UTXO data is
+// specified in the psbt package. This check is necessary because of 2 reasons.
+// The sighash calculation is now different for witnessV0 and witnessV1 inputs
+// this means we need to check the previous output pkScript for the specific
+// type and the second reason is that the sighash calculation for taproot inputs
+// include the previous output pkscripts.
+func InputsReadyToSign(packet *Packet) error {
+	VerifyInputOutputLen(packet, true, true)
+
+	for i := range packet.UnsignedTx.TxIn {
```

Comment:
I guess than wherever this is called we have already checked that `UnsignedTx` is not `nil`?